*** sqlbulkcopy-cannot-access-table ***

 After reading in an excel-sheet (to transferTable), I want to add that data to a new table (destinationTable) using SqlBulkCopy, but I'm getting the error: 
 
 I've tried using the default tablename and using square brackets, but that didn't work. 
 Any suggestions? 
 
 
 My issue was a bit different, turns out my table name was a reserved keyword in SQL so I had to do the following: 
 
 Where   is the target schema and   the target table name 
 From the  documentation 
 
 DestinationTableName is a three-part name [database].[owningschema].[name]. You can qualify the table name with its database and owning schema if you choose. However, if the table name uses an underscore ("_") or any other special characters, you must escape the name using surrounding brackets as in ([database].[owningschema].[name_01]) 
 
 
 Check that user that connects to db has  
 
 as suggested in answer by Jhilden on  MSDN forum . 
 
 It seems that the user who executes this code don't have proper access to the database.
 * Check so that the user got access.
 * Check the connectionstring your using to connect to the database. 
 
 I recently ran into this same error and came across this post while googling for an answer.  I was able to solve the problem by giving the user that is executing the bulk copy command insert and select permissions on the destination table.
Originally I had only granted insert permission to the user and got the 'Cannot access destination table' error. 
 
 Interestingly, this also happens if you have a table name which is purely numeric. Start the table name with one or more alpha characters and it works just fine. 
 
 In my case, it's not a permission problem, but a special char in the table name problem ( parenthesis and & ). 
 Hope this helps 
 
 Bulkcopy expects the table to exists in the database. Also you should have access to this database or table. 
 
 Andrij Ferents answer  is valid.  
 The destination table must exist before calling  . It is a common newbie mistake. 
 
 In my case, the problem was because of an existing Identity column  
 